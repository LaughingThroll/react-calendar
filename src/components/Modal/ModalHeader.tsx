import React from "react"

interface IModalHeader {
  title?: string
  countDays?: number
}

const ModalHeader = ({ title, countDays = 0 }: IModalHeader) => {
  return (
    <div className="modal-header">
      <h4 className="modal-header__title">{title}</h4>
      <div className="modal-header__days">{countDays} Days</div>
    </div>
  )
}

export default ModalHeader
