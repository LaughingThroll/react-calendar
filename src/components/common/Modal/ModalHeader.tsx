import React from 'react'

interface IModalHeader {
  className?: string[]
}

const ModalHeader: React.FC<IModalHeader> = ({ className = [], children }) => {
  return <div className={`modal-header ${className.join(' ')}`}>{children}</div>
}

export default ModalHeader
