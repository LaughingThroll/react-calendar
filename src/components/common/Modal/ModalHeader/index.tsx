import React from 'react'
import styles from './modalHeader.module.scss'
import { TChildren } from '../../../../types/utilsTypes'

interface ModalHeaderProps {
  className?: string[]
  children?: TChildren
}

const { modalHeader } = styles

const ModalHeader: React.FC<ModalHeaderProps> = ({ className = [], children }) => {
  return <div className={`${modalHeader} ${className.join(' ')}`}>{children}</div>
}

export default React.memo(ModalHeader)
