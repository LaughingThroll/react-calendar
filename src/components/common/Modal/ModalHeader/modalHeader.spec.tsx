import React from 'react'
import { shallow } from 'enzyme'
import ModalHeader from './'

describe('ModalHeader render', () => {
  it('render default', () => {
    const tree = shallow(<ModalHeader />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(1)
  })

  it('render with children', () => {
    const tree = shallow(<ModalHeader children={<span>Test</span>} />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('span')).toHaveLength(1)
  })
})

describe('ModalHeader props', () => {
  it('component wiht className', () => {
    const tree = shallow(<ModalHeader className={['hello']} />)

    expect(tree.find('div').hasClass('hello')).toBe(true)
  })
})
