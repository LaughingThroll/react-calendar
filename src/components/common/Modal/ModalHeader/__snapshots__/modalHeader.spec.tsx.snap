// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader render render default 1`] = `
<div
  className="modalHeader "
/>
`;

exports[`ModalHeader render render with children 1`] = `
<div
  className="modalHeader "
>
  <span>
    Test
  </span>
</div>
`;
